contracts/src/v0.8/dev/keeper2_0/KeeperRegistry2_0.sol

```diff
@@ -309,37 +434,170 @@ contract KeeperRegistry2_0 is
+    if (signers.length != transmitters.length || signers.length <= 3 * f) revert IncorrectNumberOfSigners();
+    if (onchainConfig.length != 0) revert OnchainConfigNonEmpty();
+
+    // remove any old signer/transmitter addresses
+    uint256 oldLength = s_signersList.length;
+    address signer;
+    address transmitter;
+    for (uint256 i = 0; i < oldLength; i++) {
+      signer = s_signersList[i];
+      transmitter = s_transmittersList[i];
+      delete s_signers[signer];
+      // Do not delete the whole transmitter struct as it has balance information stored
+      s_transmitters[transmitter].active = false;
+    }
+    delete s_signersList;
+    delete s_transmittersList;
+
+    // add new signer/transmitter addresses
+    for (uint256 i = 0; i < signers.length; i++) {
+      if (s_signers[signers[i]].active) revert RepeatedSigner();
+      s_signers[signers[i]] = Signer({active: true, index: uint8(i)});
+
+      if (s_transmitters[transmitters[i]].active) revert RepeatedTransmitter();
+      s_transmitters[transmitters[i]].active = true;
+      s_transmitters[transmitters[i]].index = uint8(i);
+    }
+    s_signersList = signers;
+    s_transmittersList = transmitters;
+    s_hotVars.f = f;
+    s_offchainConfigVersion = offchainConfigVersion;
+    s_offchainConfig = offchainConfig;
+
+    _computeAndStoreConfigDigest(
+      signers,
+      transmitters,
+      f,
+      abi.encode(
+        OnChainConfig({
+          paymentPremiumPPB: s_hotVars.paymentPremiumPPB,
+          flatFeeMicroLink: s_hotVars.flatFeeMicroLink,
+          checkGasLimit: s_storage.checkGasLimit,
+          stalenessSeconds: s_hotVars.stalenessSeconds,
+          gasCeilingMultiplier: s_hotVars.gasCeilingMultiplier,
+          minUpkeepSpend: s_storage.minUpkeepSpend,
+          maxPerformGas: s_storage.maxPerformGas,
+          maxCheckDataSize: s_storage.maxCheckDataSize,
+          maxPerformDataSize: s_storage.maxPerformDataSize,
+          fallbackGasPrice: s_storage.fallbackGasPrice,
+          fallbackLinkPrice: s_storage.fallbackLinkPrice,
+          transcoder: s_storage.transcoder,
+          registrar: s_storage.registrar
+        })
+      ),
+      offchainConfigVersion,
+      offchainConfig
+    );
+  }
+
   /**
    * @notice updates the configuration of the registry
-   * @param config registry config fields
-   */
-  function setConfig(Config memory config) public onlyOwner {
-    if (config.maxPerformGas < s_storage.maxPerformGas) revert GasLimitCanOnlyIncrease();
+   * @param onChainConfig registry config fields
+   */
+  function setOnChainConfig(OnChainConfig memory onChainConfig) public onlyOwner {
+    if (onChainConfig.maxPerformGas < s_storage.maxPerformGas) revert GasLimitCanOnlyIncrease();
+    if (onChainConfig.maxCheckDataSize < s_storage.maxCheckDataSize) revert MaxCheckDataSizeCanOnlyIncrease();
+    if (onChainConfig.maxPerformDataSize < s_storage.maxPerformDataSize) revert MaxPerformDataSizeCanOnlyIncrease();
```

Comment:
I added these two for consistency with gasLimitCanIncrease but have mixed feelings about it. Can remove all of them if folks think its too much of an overhead